### DISCUSSION TITLE
Provide authentication for Twitch to avoid advertisements?

### DISCUSSION THREAD

**Discussion by Coalbus** [Tzahi12345/YoutubeDL-Material#382] [Q&A] [answered]
Hello!

Very impressed with YoutubeDL-Material so far, as well as your enthusiasm and responsiveness!

I am a bit simple, so please forgive me if this is a dumb/obvious question. Is there a way to provide my Twitch credentials such that when I download a live stream it does not show the pre-roll ads for channels that I am subbed to? (rather, the static screen in the downloaded video where ads would be played)

I see in the settings there's an "Args" section and there's an "authentication" setting under "Search by Category", but to be honest I can't figure out how this would be implemented specifically for Twitch. Or would this be done by the "Twitch API Key" section?

Sorry again for my lack of knowledge on such things, and thank you for your time.

**Comment by Tzahi12345** [comment, marked as answer]
Hi sorry for the delay as I took a bit of a hiatus (ironically right as you commend me for being on top of things)

When I download twitch videos I don't get ads, is this something you see from your end? Perhaps you can try using [cookies](https://apple.stackexchange.com/questions/349697/how-to-use-youtube-dl-cookies/349759#349759) to avoid this.

**Reply by Tzahi12345** [reply]
Just found this comment https://github.com/ytdl-org/youtube-dl/issues/23337#issuecomment-567768049

**Reply by Coalbus** [reply]
Thanks! And no worries, life comes first.

I did manage to figure it out eventually. I couldn't get the cookies command to work, but I took another look at the options in the web UI and realized that I somehow overlooked the "Use Cookies" option.

**So, for anyone that comes here looking for the same thing in the future:**
1. Obtain the cookie for the desired website. I use a Chrome extension called "Get cookies.txt" to generate a cookies file.
2. Save the cookies.txt file somewhere easy to find
3. Open the YoutubeDL-Material web interface
4. Click the 3 vertical dots in the top right
5. Click Settings
6. Go to the Advanced tab
7. Put a check in "Use Cookies" then click "Set Cookies" and select the cookies txt file you created earlier, then click Upload.
8. Save and exit the Settings screen
9. Done!

This seems to work for just about any website that Youtube-DL itself can download from. Keep in mind that it can only handle cookies for 1 website at a time, so if you regularly download from say YouTube and Twitch, you'll need to keep a cookies file handy for both sites and switch between them if needed.

**Reply by Tzahi12345** [reply]
That's the unfortunate thing with cookies, though we could do it so we choose cookies based on the website that's being used. Shouldn't be too difficult, feel free to create an issue for this.
